### PR TITLE
Generate interface instead of intersection for object models

### PR DESCRIPTION
**Pull request by deyaaeldeen**
Generated object models will change from:

```ts
export type CatAPTrue = PetAPTrue & {
  friendly?: boolean;
};
```
to
```ts
export interface CatAPTrue extends PetAPTrue {
  friendly?: boolean;
}
```

This change addresses feedback received during review for the new API for the text analytics library where a lot of the generated models are exported as part of the public surface. Instead of recreating all those models as interfaces manually, this PR updates the code generator to generate them as interfaces from the get-go.

### CODE REVIEW

File: test/integration/generated/textanalytics/src/models/index.ts [hunk 1]
```diff
@@ -525,51 +525,54 @@ export interface SentenceAssessment {
...
+  extends AnalyzeJobMetadata,
+    TasksState,
+    Pagination {
+  errors?: TextAnalyticsError[];
+  /** if includeStatistics=true was specified in the request this field will contain information about the request payload. */
+  statistics?: TextDocumentBatchStatistics;
+}
 
-export type TasksStateTasksDetails = TaskState;
+export interface TasksStateTasksDetails extends TaskState {}
```
Comment (deyaaeldeen) [inline test/integration/generated/textanalytics/src/models/index.ts:551]: perhaps generate an interface only if the list of properties is not empty.

Comment (joheredi) [reply test/integration/generated/textanalytics/src/models/index.ts:551]: I think practically it is the same as having the alias. So I think I prefer the simplicity of the new code over handling the case to generate the alias.

The ideal thing may be to not generate it at all and give anything that has a `TasksStateTasksDetails` type  `TaskState` instead. But I think that work would be a lot more involved.

I'm fine with the new generated interface

**Review by joheredi — APPROVED** [review]
I think this looks great! Thanks for taking the time to send out the fix.

@qiaozha would you mind taking a look as well please?

**Review by qiaozha — APPROVED** [review]
LGTM

**Review by witemple-msft — APPROVED** [review]
This will _extremely significantly_ improve the quality of generated documentation for packages that export generated types.

Thank you @deyaaeldeen for looking into this and LGTM.

File: test/integration/generated/textanalytics/src/models/index.ts (hunk 1 above)
Comment (deyaaeldeen) [reply test/integration/generated/textanalytics/src/models/index.ts:551]: Thanks for your valuable perspective! I agree with all what you said :)

**Pull request merged**